dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/RequestItemRepository.java

```diff
@@ -282,33 +328,45 @@ public RequestItemRest put(Context context, HttpServletRequest request,
         return rir;
     }
 
-    @Override
-    public Class<RequestItemRest> getDomainClass() {
-        return RequestItemRest.class;
-    }
-
     /**
-     * Generate a link back to DSpace, to act on a request.
      *
-     * @param token identifies the request.
-     * @return URL to the item request API, with /request-a-copy/{token} as the last URL segments
-     * @throws URISyntaxException passed through.
-     * @throws MalformedURLException passed through.
+     * @param accessToken
+     * @return
      */
-    public String getLinkTokenEmail(String token)
-            throws URISyntaxException, MalformedURLException {
-        final String base = configurationService.getProperty("dspace.ui.url");
-
-        // Construct the link, making sure to support sub-paths
-        URIBuilder uriBuilder = new URIBuilder(base);
-        List<String> segments = new LinkedList<>();
-        if (StringUtils.isNotBlank(uriBuilder.getPath())) {
-            segments.add(StringUtils.strip(uriBuilder.getPath(), "/"));
+    @PreAuthorize("permitAll()")
+    @SearchRestMethod(name = "byAccessToken")
+    public RequestItemRest findByAccessToken(@Parameter(value = "accessToken", required = true) String accessToken) {
+
+        // Send 404 NOT FOUND if access token is null
+        if (StringUtils.isBlank(accessToken)) {
+            throw new ResourceNotFoundException("No such accessToken=" + accessToken);
+        }
+
+        // Get the current context and request item
+        Context context = obtainContext();
+        RequestItem requestItem = requestItemService.findByAccessToken(context, accessToken);
+
+        // Send 404 NOT FOUND if request item is null
+        if (null == requestItem) {
+            throw new ResourceNotFoundException("No such request item for accessToken=" + accessToken);
         }
-        segments.add("request-a-copy");
-        segments.add(token);
 
-        // Build and return the URL from segments (or throw exception)
-        return uriBuilder.setPathSegments(segments).build().toURL().toExternalForm();
+        // Send 403 FORBIDDEN if request access has not been granted or access period is null or in the past
+        if (!requestItem.isAccept_request() ||
+                requestItem.getAccess_expiry() == null ||
+                requestItem.getAccess_expiry().isBefore(Instant.now())) {
+            throw new AccessDeniedException("Access has not been granted for this item request");
```

Comment:
Would it be possible to still expose expired `RequestItem`s, maybe with a `boolean expired` flag based on the `isBefore` above?

This would make things a lot more clear for people that try to visit an expired link; right now they will just see a login page.

The Bitstreams would still be "safe" ~ `authorizeAccessByAccessToken`

(related to https://github.com/DSpace/dspace-angular/pull/3984#discussion_r2016742001 and https://github.com/DSpace/dspace-angular/pull/3984#discussion_r2016745272)

Reply:
I've done this, and changed the frontnend to display different error alerts depending on the results. Thanks!